nomad/alloc_endpoint.go

```diff
@@ -370,3 +378,12 @@ func (a *Alloc) UpdateDesiredTransition(args *structs.AllocUpdateDesiredTransiti
 	reply.Index = index
 	return nil
 }
+
+func (a *Alloc) validateTLSCertificate() error {
+	if a.srv.config.TLSConfig == nil || !a.srv.config.TLSConfig.VerifyServerHostname {
+		return nil
+	}
+
+	expected := fmt.Sprintf("client.%s.nomad", a.srv.Region())
+	return a.ctx.ValidateCertificateForName(expected)
+}
```

Comment:
These became quite repetitive, but I'm not sure if there's a better place to have them. I thought about moving it to the server and pass the RPC context, but I'm not sure if it's the best place.

Reply:
Indeed, seems like this function could be extracted into a generic function, e.g. 

```
func ValidateTLSCertificate(ctx Context, config *TLSConfig, verifyServerHostname bool, region string) {
...
}
```

and just put that under `helpers/` somewhere?

Reply:
I'm trying to imagine what all this would look like if we could have handler wrappers like `gin` or other web frameworks. Probably not anywhere near possible without a good bit of refactoring though.

Reply:
Ah good call. I moved them to a set of helper functions in `nomad/util.go`